assets/src/hooks/v2/use_simulation_api_response.tsx

```diff
@@ -0,0 +1,161 @@
+import { WidgetData } from "Components/v2/widget";
+import useDriftlessInterval from "Hooks/use_driftless_interval";
+import React, { useEffect, useState } from "react";
+import { ApiResponse } from "./use_api_response";
+
+const MINUTE_IN_MS = 60_000;
+
+interface RawResponse {
+  data: WidgetData | null;
+  force_reload: boolean;
+  disabled: boolean;
+}
+
+type SimulationApiResponse =
+  // The request was successful.
+  {
+    fullPage: ApiResponse;
+    flexZone: ApiResponse;
+  };
+
+const FAILURE_RESPONSE: SimulationApiResponse = {
+  fullPage: { state: "failure" },
+  flexZone: { state: "failure" },
+};
```

Comment:
Is the different shape of response data the main difference between `use_api_response` and this module? Wondering if we could avoid having a copy-pasted `useApiResponse`—I'm worried about code drift over time...

Factoring out the common code could be an interesting exercise, let me know if you'd like to pair on it 🤓

Reply:
The response shape was definitely what put me on this path at first. In the very beginning of this branch, I tried using `use_api_response` for simulations, but the amount of refactoring that was needed seemed dangerous given that we want the current version of the hook to be completely unchanged for existing endpoints. In my mind, I have felt like all things simulations should be siloed into their own hooks/functions/components so that we don't change something on live screens by mistake. Open to any suggestions on this, though.

Reply:
I think if the difference is only the response data + the logic that parses it, and we could have it be handled differently in the two cases by passing a callback function with the parsing logic, it would still be good to define the common logic around firing requests in one place. There's a clean separation between "fire requests, parse responses using [handler], handle the responses" and the "[handler]" part, as long as we strictly define the expected inputs and outputs of "[handler]".

In fact you've basically done that as far as I can tell—you call `rawResponseToSimulationResponse` vs `rawResponseToApiResponse`.